code/plugins/ccavenue/ccavenue.php

```diff
@@ -176,13 +176,13 @@ public function onTP_GetHTML($vars)
 
 		// Notify URL set for notify url parameter of payment gateway
 		$gatewaydata['notify_url']      = $vars->notify_url;
-		$gatewaydata['billing_name']    = $vars->userInfo['firstname'] . ' ' . $vars->userInfo['lastname'];
+		$gatewaydata['billing_name']    = $vars->user_firstname . ' ' . $vars->user_lastname;
```

Comment:
@vaibhavsTekdi $vars is the value we get from the extension we can add $vars->user_firstname and $vars->user_firstname from the extension itself.

If we change or remove $vars->userInfo['firstname'] it might break the backward compatibility and might stop working for other extensions